src/settings.ts

```diff
@@ -42,7 +49,11 @@ export async function getSettings() {
   const fileHandle = await open(filePath, 'r')
   try {
     const settings = await fileHandle?.readFile({ encoding: 'utf-8' })
-    return JSON.parse(settings) as AppSettings
+    const currentSettings = JSON.parse(settings) as AppSettings
```

Comment:
You have a schema defined, should be able to use it here to avoid the type assertion
```suggestion
    const currentSettings = AppSettingsSchema.parse(settings)
```